Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/use-contests.tsx

```diff
@@ -95,24 +109,57 @@ const ContestsProvider = ({ children }: IContestsProviderProps) => {
         () => strategiesAreLoaded && categoriesAreLoaded
             ? generateStatusFilters()
                 .concat(generateCategoryFilters(categories))
-                .concat(generateStrategyFilters(strategies))
-            : [],
+                .concat(generateStrategyFilters(strategies)) as IFilter[]
+            : [] as IFilter[],
         [ categories, categoriesAreLoaded, strategies, strategiesAreLoaded ],
     );
 
+    const possibleSortingTypes = useMemo(
+        () => generateSortingStrategy() as unknown as ISort[],
+        [],
+    );
+    
     const filters = useMemo(
-        () => collectFilters(params, possibleFilters),
+        () => collectParams(params, possibleFilters, DEFAULT_FILTER_TYPE, DEFAULT_STATUS_FILTER_TYPE),
         [ params, possibleFilters ],
     );
 
+    const sortingTypes = useMemo(
+        () => collectParams(params, possibleSortingTypes, DEFAULT_SORT_FILTER_TYPE, DEFAULT_SORT_TYPE),
+        [ params, possibleSortingTypes ],
+    );
+
     const currentPage = useMemo(
         () => collectCurrentPage(params),
         [ params ],
     );
 
     const clearFilters = useCallback(
-        () => clearParams(),
-        [ clearParams ],
+        (types: FilterType[], filterTypeId: number) => {
```

Comment:
Why `clearFilters` gets an arg `types`?
This should be internal to this hook.
Otherwise, external code should have the information about `filter types` which is strange.

Also, clear filters should only clear the `filters`, i.e. not `sort` as **`sort` is not a filter**

IMO, this function should have no arguments.

The functions should be 2 - `clearFilter` and `clearSorts`, each doing its cohesive task. Both not taking any arguments, as they should remove the respected part internally.

Reply:
I made another function - clearSorts, which now also as clearFilter does not accept any arguments and each of them is used in the respective component.